apps/framework-cli/src/cli/display/infrastructure.rs

```diff
@@ -163,26 +296,161 @@ pub fn show_olap_changes(olap_changes: &[OlapChange]) {
+                                    before_str.push_str(" NOT NULL");
+                                } else {
+                                    before_str.push_str(" NULLABLE");
+                                }
+                                if let Some(ref default) = before.default {
+                                    before_str.push_str(&format!(" DEFAULT={}", default));
+                                }
+
+                                // Build after state
+                                let mut after_str = format_column_type(&after.data_type);
+                                if after.required {
+                                    after_str.push_str(" NOT NULL");
+                                } else {
+                                    after_str.push_str(" NULLABLE");
+                                }
+                                if let Some(ref default) = after.default {
+                                    after_str.push_str(&format!(" DEFAULT={}", default));
+                                }
+
+                                // Add unique constraint changes if any
+                                let mut extra_changes = Vec::new();
+                                if before.unique != after.unique {
+                                    if after.unique {
+                                        extra_changes.push("now unique".to_string());
+                                    } else {
+                                        extra_changes.push("no longer unique".to_string());
+                                    }
+                                }
+
+                                if extra_changes.is_empty() {
+                                    format!("  ~ {}: {} -> {}", before.name, before_str, after_str)
+                                } else {
+                                    format!(
+                                        "  ~ {}: {} -> {}, {}",
+                                        before.name,
+                                        before_str,
+                                        after_str,
+                                        extra_changes.join(", ")
+                                    )
+                                }
+                            }
+                        }
                     };
                     details.push(change_line);
                 }
             }
 
+            // Check for primary key changes (causes table recreation)
+            let before_pk: Vec<String> = before
+                .primary_key_columns()
+                .into_iter()
+                .map(|s| s.to_string())
+                .collect();
+            let after_pk: Vec<String> = after
+                .primary_key_columns()
+                .into_iter()
+                .map(|s| s.to_string())
+                .collect();
+
+            if before_pk != after_pk {
+                details.push("Primary key changes (requires table recreation):".to_string());
```

Comment:
this concern should be addressed by `ClickHouseTableDiffStrategy`

Reply:
Yeah - this is to inform the deletion / creation? or are you saying that it will be displayed to the user anyway through the delete and create operations?